pymatgen/electronic_structure/bandstructure.py

```diff
@@ -819,143 +954,35 @@ def apply_scissor(self, new_band_gap):
                             # spin_index = Spin.down
             old_dict = self.as_dict()
             shift = new_band_gap
-            for spin in old_dict['bands']:
-                for k in range(len(old_dict['bands'][spin])):
-                    for v in range(len(old_dict['bands'][spin][k])):
+            for spin in old_dict["bands"]:
+                for k in range(len(old_dict["bands"][spin])):
+                    for v in range(len(old_dict["bands"][spin][k])):
                         if k >= max_index:
-                            old_dict['bands'][spin][k][v] = \
-                                old_dict['bands'][spin][k][v] + shift
+                            old_dict["bands"][spin][k][v] = (
+                                old_dict["bands"][spin][k][v] + shift
+                            )
         else:
 
-            shift = new_band_gap - self.get_band_gap()['energy']
+            shift = new_band_gap - self.get_band_gap()["energy"]
             old_dict = self.as_dict()
-            for spin in old_dict['bands']:
-                for k in range(len(old_dict['bands'][spin])):
-                    for v in range(len(old_dict['bands'][spin][k])):
-                        if old_dict['bands'][spin][k][v] >= \
-                                old_dict['cbm']['energy']:
-                            old_dict['bands'][spin][k][v] = \
-                                old_dict['bands'][spin][k][v] + shift
-            old_dict['efermi'] = old_dict['efermi'] + shift
+            for spin in old_dict["bands"]:
+                for k in range(len(old_dict["bands"][spin])):
+                    for v in range(len(old_dict["bands"][spin][k])):
+                        if old_dict["bands"][spin][k][v] >= old_dict["cbm"]["energy"]:
+                            old_dict["bands"][spin][k][v] = (
+                                old_dict["bands"][spin][k][v] + shift
+                            )
+            old_dict["efermi"] = old_dict["efermi"] + shift
         return self.from_dict(old_dict)
 
     def as_dict(self):
         """
         Json-serializable dict representation of BandStructureSymmLine.
         """
-
-        d = {"@module": self.__class__.__module__,
-             "@class": self.__class__.__name__,
-             "lattice_rec": self.lattice_rec.as_dict(), "efermi": self.efermi,
-             "kpoints": []}
-        # kpoints are not kpoint objects dicts but are frac coords (this makes
-        # the dict smaller and avoids the repetition of the lattice
-        for k in self.kpoints:
-            d["kpoints"].append(k.as_dict()["fcoords"])
+        d = super().as_dict()
         d["branches"] = self.branches
```

Comment:
If as_dict serves only the purpose of storing what needed for rebuild the object, yes, I'd agree. 
Though, I was leaving this because in the previous version the dict contains explicitly the branches as well. 
I do not know if somewhere the BandstructureSymmLine dict is used as a dict and branches are needed.